src/transformStateWithClones.js

```diff
@@ -7,7 +7,57 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const NewArray = [];
+  let massive = { ...state };
+
+  for (let numArr = 0; numArr < actions.length; numArr++) {
+    const action = Object.assign({}, actions[numArr]);
+
+    for (const key in action) { // loop all types in array 'action'
+      if (action[key] === 'addProperties') {
+        const state1 = { ...massive };
+
+        Object.assign(state1, action.extraData);
+        NewArray.push(state1);
+
+        for (const b in massive) {
+          delete massive[b];
+        }
+        massive = { ...state1 };
+      } else if (action[key] === 'clear') {
+        const state2 = { ...massive };
+
+        for (const x in state2) {
+          delete state2[x];
+        };
+
+        NewArray.push(state2);
+
+        for (const b in massive) {
+          delete massive[b];
+        }
+        massive = { ...state2 };
```

Comment:
Why is it needed? You need only to add properties in this 'if'. Please, revise your solution and remove unnecessary code from other 'if's - leave only what is needed for this specific 'if'